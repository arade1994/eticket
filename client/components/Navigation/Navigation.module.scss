.navigationContainer {
  height: 100%;
  position: sticky;
  top: 50px;

  & .navigation {
    height: 85vh;
    background: rgb(242, 244, 245);
    border: 0px solid rgba(7, 64, 82, 1);
    border-left: none;
    border-top-right-radius: 5px;
    box-shadow: 0px 0px 3px rgb(161, 163, 163);
    display: flex;
    flex-direction: column;
    gap: 5px;
    justify-content: flex-start;
    padding: 15px;

    & a {
      color: rgb(0, 65, 85);
      font-size: 18px;
      border-radius: 5px;
      text-decoration: none;

      &:focus {
        color: rgb(131, 182, 182);
      }

      &:hover {
        text-decoration: underline;
      }
    }
  }
}

.marginTopAuto {
  margin-top: auto;
}
