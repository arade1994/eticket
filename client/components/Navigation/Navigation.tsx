import Link from "next/link";
import cx from "classnames";

import classes from "./Navigation.module.scss";
import { NAVIGATION_LINKS } from "../../utils/constants";

const Navigation = () => {
  return (
    <aside className={classes.navigationContainer}>
      <nav className={classes.navigation}>
        {NAVIGATION_LINKS.map(({ href, label }, index) => (
          <Link
            key={href}
            className={cx(
              index === NAVIGATION_LINKS.length - 1 ? classes.marginTopAuto : ""
            )}
            href={href}
          >
            {label}
          </Link>
        ))}
      </nav>
    </aside>
  );
};

export default Navigation;
