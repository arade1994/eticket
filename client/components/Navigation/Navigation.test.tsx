import { render, screen } from "@testing-library/react";
import { describe, expect, test } from "vitest";
import Navigation from "./Navigation";

describe("<Navigation />", () => {
  test("it should contain all the navigation links by default", () => {
    const { baseElement } = render(<Navigation />);

    expect(baseElement).toMatchSnapshot();

    const ticketsLink = screen.getByText("Tickets");
    const ordersLink = screen.getByText("Orders");
    const usersLink = screen.getByText("Users");
    const signOutLink = screen.getByText("Sign out");

    expect(ticketsLink).toBeDefined();
    expect(ordersLink).toBeDefined();
    expect(usersLink).toBeDefined();
    expect(signOutLink).toBeDefined();
  });
});
