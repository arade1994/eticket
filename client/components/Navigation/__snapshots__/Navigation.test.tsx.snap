// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`<Navigation /> > it should contain all the navigation links by default 1`] = `
<body>
  <div>
    <aside
      class="_navigationContainer_1558d5"
    >
      <nav
        class="_navigation_1558d5"
      >
        <a
          class=""
          href="/"
        >
          Tickets
        </a>
        <a
          class=""
          href="/orders"
        >
          Orders
        </a>
        <a
          class=""
          href="/users"
        >
          Users
        </a>
        <a
          class="_marginTopAuto_1558d5"
          href="/auth/signout"
        >
          Sign out
        </a>
      </nav>
    </aside>
  </div>
</body>
`;
